src/main/java/org/folio/rest/impl/TenantRulesImpl.java

```diff
@@ -178,4 +181,16 @@ private String validateRule(Rule entity) {
     return errorMessage;
   }
 
+  private String validateRuleOnPut(Rule entity) {
+    if (StringUtils.isEmpty(entity.getId()) || StringUtils.isEmpty(entity.getRuleId())) {
+      logger.debug("Rule id cannot be null or empty");
```

Comment:
This implementation requires that the client always sets both id and ruleId to the same value. If this is intended it is a breaking change and the ModuleDescriptor interface version needs to be incremented.

My suggestion is a non-breaking solution:
The client can set one of the two properties (id, ruleId) and TenantRulesImpl will automatically copy the value from the that property into the empty property.

Reply:
changed _validateRuleOnPut_, now entity with _ruleId_ or _id_ is valid